packages/framework/client-logger/external-telemetry/src/container/telemetryProducer.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IContainer,
+	IContainerEvents,
+	type ICriticalContainerError,
+} from "@fluidframework/container-definitions";
+import {
+	ContainerConnectedTelemetry,
+	ContainerTelemetryEventNames,
+	IContainerTelemetry,
+	ContainerDisconnectedTelemetry,
+	ContainerClosedTelemetry,
+	ContainerAttachingTelemetry,
+	ContainerAttachedTelemetry,
+	type ContainerTelemetryEventName,
+} from "./containerTelemetry";
+import { ContainerSystemEventName, ContainerSystemEventNames } from "./containerSystemEvents";
+
+/**
+ * This class produces {@link IContainerTelemetry} from raw container system events {@link IContainerEvents}.
+ * The class contains different helper methods for simplifying and standardizing logic for adding additional information necessary
+ * to produce different {@link IContainerTelemetry}.
+ *
+ * @internal
+ */
+export class ContainerEventTelemetryProducer {
+	constructor(private container: IContainer) {}
+
+	public produceTelemetry(
+		eventName: ContainerSystemEventName,
+		payload?: any,
+	): IContainerTelemetry | undefined {
+		let telemetry: IContainerTelemetry | undefined = undefined;
```

Comment:
Nit: we have a mix of approaches here; the first case block directly returns something, while all other ones just set the variable and break, to let the final `return` do its thing. I'd consolidate on all case blocks directly returning the corresponding result, we don't even need to assign them to any local variables. That'd make the function half-ish the size :).

Reply:
Updated.